### DISCUSSION TITLE
Can AsyncReadWriteLock support acquiring an upgradable read lock asynchronously?

### DISCUSSION THREAD

**Discussion by herebebeasties** [dotnet/dotNext#216] [Q&A] [answered]
Is there any way to simulate / implement an `AsyncReadWriteLock.AcquireUpgradeableReadLockAsync(...)` method, which is not currently present?
I see that `AsyncLock.ReadLock(...)` doesn't take an `upgradeable` parameter, unlike its synchronous counterpart `Lock.ReadLock()`.

**Comment by sakno** [comment, marked as answer]
Hi @herebebeasties , you can call `AcquireReadLockAsync` and then `TryUpgradeToWriteLock` or `UpgradeToWriteLockAsync`

**Reply by herebebeasties** [reply]
Thanks! I should have read the docs better.
